src/Autocomplete.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { ChangeEvent } from 'react';
+
+import { Field, FieldRenderProps, FieldProps } from 'react-final-form';
+
+import TextField, { TextFieldProps as MuiTextFieldProps } from '@material-ui/core/TextField';
+import { default as MuiAutocomplete, AutocompleteProps as MuiAutocompleteProps } from '@material-ui/lab/Autocomplete';
+
+export type AutocompleteOption = {
+	[key: string]: any;
+};
+
+interface AutocompleteProps extends Partial<MuiAutocompleteProps> {
+	name: string;
+	label: string;
+	getOptionValue: (option: any) => any;
+	options: AutocompleteOption[];
+	required?: boolean;
+	fieldProps?: FieldProps<any, any>;
+	textFieldProps?: Partial<MuiTextFieldProps>;
+}
+
+export const Autocomplete = (props: AutocompleteProps) => {
+	const { name, fieldProps, ...rest } = props;
+
+	return (
+		<Field
+			name={name}
+			render={fieldRenderProps => <AutocompleteWrapper {...fieldRenderProps} {...rest} />}
+			{...fieldProps}
+		/>
+	);
+};
+
+interface MuiRffAutocompleteProps extends FieldRenderProps<MuiTextFieldProps, HTMLElement> {
+	label: string;
+	textFieldProps?: Partial<MuiTextFieldProps>;
+	getOptionValue: (option: any) => any;
+}
+
+const AutocompleteWrapper = (props: MuiRffAutocompleteProps) => {
+	const {
+		input: { name, onChange, value, multiple, ...restInput },
+		meta,
+		label,
+		textFieldProps,
+		getOptionValue,
+		...rest
+	} = props;
+
+	const showError = ((meta.submitError && !meta.dirtySinceLastSubmit) || meta.error) && meta.touched;
+
+	return (
+		<MuiAutocomplete
+			{...rest}
+			multiple={multiple}
+			onChange={
+				multiple
+					? (_e: ChangeEvent<{}>, values: any[]): void => {
+							onChange(values.map(getOptionValue));
+					  }
+					: (_e: ChangeEvent<{}>, option: any): void => {
+							option ? onChange(getOptionValue(option)) : onChange(undefined);
+					  }
+			}
+			renderInput={params => (
+				<TextField
+					label={label}
+					{...params}
+					{...restInput}
+					{...textFieldProps}
```

Comment:
I'd put these 3 `{...}` at the end so that things like margin=normal and variant=standard can get overridden.

Reply:
Yep that's fair. I think it will cause some Typing issues that need to be fixed  - so not quite as simple as shifting a few lines. 👍

Reply:
Really? The order of component properties shouldn't affect types... or at least that hasn't been my experience. It just affects the overrides.

Reply:
yeah it's related to `textFieldProps` being set to `Partial<MuiTextFieldProps>`, and how that handles overwriting the `variant` prop. It's a bit weird, but should just be a matter of getting the types correct.